### PR TITLE
disable and enable temperature training in MoT

### PR DESCRIPTION
**Pull request by Simontwice**
# Description

# Checklist
   - [x] Is this PR focused on a single feature without the possibility of being divided into smaller PRs?
   - [x] Are all variables set with logical defaults that are backward compatible?
   - [x] I attached link to neptune if it was necessary
   - [x] Have you successfully executed the linter and tests?
   - [x] Are all functions concise and don't require splitting?
   - [x] Is there documentation for arguments and complex logic?
   - [x] Are there no temporary solutions in the code? Should a task be added to Trello to refine the code?
   - [x] Are all functions placed in appropriate files?
   - [x] Is the PR name meaningful and descriptive?
   - [x] Is PR description provided, or unnecessary?
  
Other questions:
   - [x] If I made changes to the requirements or anything related to the Singularity image I did generate new image and upload a new image to the clusters
   - [x] If this change is significant enough I notified all individuals working with this repository (@channel in slack channel)
   - [x] If there is a need for a second reviewer I requested additional review 

# Reviewer's checklist:
   - [x] Does the Neptune link work and does it actually compare the situation before and after the PR?
   - [x] Do I understand the code?
   - [ ] Do I think another reviewer is not needed or I commented that this PR need another reviewer?

### CODE REVIEW

File: research/conditional/utils/layer_manager.py [hunk 1]
```diff
@@ -63,6 +70,15 @@ def log(self, step):
                             title=logging_name, iteration=step, data=data
                         )
 
+    def manage_misc(self, step):
+        is_learning_temperature = False
+        if step > self.steps_until_start_temperature_learn:
+            is_learning_temperature = True
```
Comment (crewtool) [inline research/conditional/utils/layer_manager.py:76]:
```suggestion
        is_learning_temperature = step > self.steps_until_start_temperature_learn
```

File: research/conditional/utils/layer_manager.py [hunk 2]
```diff
@@ -63,6 +70,15 @@ def log(self, step):
...
+        for block_name, layer in self._layers:
+            for name, param in layer.named_parameters():
+                if "temperature" in name:
+                    param.requires_grad = is_learning_temperature
```
Comment (crewtool) [inline research/conditional/utils/layer_manager.py:80]: manage_misc seems meaningless to me

File: research/conditional/utils/argparse.py [hunk 3]
```diff
@@ -114,7 +114,7 @@ def introduce_parser_arguments(
     parser.add_argument("--should_evaluate_dynamic_groupsize", action="store_true")
 
     ## used by MoE (some specific, some common)
-
+    parser.add_argument("--steps_until_start_temperature_learn", type=int, default=-1)
```
Comment (crewtool) [inline research/conditional/utils/argparse.py:117]: Value  '-1' of parameter very often means "disabled" and here  it means learnable temperature from the very first step. It is quite unintuitive for me

Comment (Simontwice) [reply research/conditional/utils/argparse.py]: adressed, thank you

File: research/conditional/utils/layer_manager.py (hunk 2 above)
Comment (Simontwice) [reply research/conditional/utils/layer_manager.py]: changed!

**Comment by crewtool** [comment]
isn't this PR a part of https://github.com/llm-random/llm-random/pull/263 ?

**Pull request closed without merging**